pkg/git/repository.go

```diff
@@ -136,10 +138,20 @@ func (o *CloneOptions) Clone(ctx context.Context) (Repository, fs.FS, error) {
 
 	r, err := clone(ctx, o)
 	if err != nil {
-		if err == transport.ErrEmptyRemoteRepository {
-			log.G(ctx).Debug("empty repository, initializing new one with specified remote")
-			r, err = initRepo(ctx, o)
+		if err != transport.ErrRepositoryNotFound && err != transport.ErrEmptyRemoteRepository {
+			return nil, nil, err
 		}
+
+		if err == transport.ErrRepositoryNotFound {
+			_, err = createRepo(ctx, o)
```

Comment:
I think there should be a `--create-repo` to the `repo bootstrap` command,  and only if it is true you should try to create the repo. Also, there are times where you cannot infer the provider, need to check that you handle it correctly and maybe add the required flags to make it work correctly so we can remove the `repo create` command

Reply:
I added a `Provider` flag on the `CreateOption`, and the plan is to expose it outside as an optional flag for the `bootstrap` command. It will be required for github enterprise servers (for example), where we can't just infer the provider form the url.